### PR TITLE
bgp: Add command for retrieving bgp routes

### PR DESCRIPTION
**Pull request by rastislavs**
This change introduces a new BGP status command: `cilium bgp routes`. BGP routes can be retrieved from either all or a specific node; in json or summary formats.

The implementation follows the existing patterns of  the `cilium bgp peers` command and mirrors the `cilium-dbg bgp routes` CLI added in https://github.com/cilium/cilium/pull/27182 .

The change also adds an alias for the existing `cilium bgp peers` command (`cilium bgp neighbors`) to align with `cilium-dbg`.
    
Vendor updates as part of the change is mostly due to the necessary BGP packet types from GoBGP.

## Examples:

Help:
```
$ cilium bgp routes -h                                                                                                                                   
Lists BGP routes from all nodes in the cluster - requires cilium >= v1.15.0

Usage:
  cilium bgp routes <available | advertised> <afi> <safi> [vrouter <asn>] [peer|neighbor <address>] [flags]

Examples:
  Get all IPv4 unicast routes available:
    cilium bgp routes available ipv4 unicast

  Get all IPv6 unicast routes available for a specific vrouter:
    cilium bgp routes available ipv6 unicast vrouter 65001

  Get IPv4 unicast routes advertised to a specific peer:
    cilium bgp routes advertised ipv4 unicast peer 10.0.0.1

Flags:
      --agent-pod-selector string   Label on cilium-agent pods to select with (default "k8s-app=cilium")
  -h, --help                        help for routes
      --node string                 Node from which BGP routes will be fetched, omit to select all nodes
  -o, --output string               Output format. One of: json, summary (default "summary")
      --wait-duration duration      Maximum time to wait for result, default 1 minute (default 1m0s)

Global Flags:
      --context string     Kubernetes configuration context
  -n, --namespace string   Namespace Cilium is running in (default "kube-system")
```

Simplest use:
```
$ cilium bgp routes                                                                                                                                                                     
(Defaulting to `available ipv4 unicast` routes, please see help for more options)

Node                              VRouter   Prefix        NextHop    Age   Attrs
bgp-cplane-dev-v4-control-plane   65001     10.1.0.0/24   0.0.0.0    55s   [{Origin: i} {Nexthop: 0.0.0.0}]                          
                                  65001     10.1.1.0/24   10.0.1.1   55s   [{Origin: i} {AsPath: 65000 65002} {Nexthop: 10.0.1.1}]   
bgp-cplane-dev-v4-worker          65002     10.1.0.0/24   10.0.2.1   55s   [{Origin: i} {AsPath: 65000 65001} {Nexthop: 10.0.2.1}]   
                                  65002     10.1.1.0/24   0.0.0.0    55s   [{Origin: i} {Nexthop: 0.0.0.0}]                     
```

Advertised routes to a specific peer of a specific node:
```
$ cilium bgp routes advertised ipv4 unicast peer 10.0.2.1 --node bgp-cplane-dev-v4-worker                                                                         
Node                       VRouter   Prefix        NextHop    Age     Attrs
bgp-cplane-dev-v4-worker   65002     10.1.1.0/24   10.0.2.2   1m37s   [{Origin: i} {AsPath: 65002} {Nexthop: 10.0.2.2}] 
```

The same as json:
```
$ cilium bgp routes advertised ipv4 unicast peer 10.0.2.1 --node bgp-cplane-dev-v4-worker -o json                                                                    
{
 "bgp-cplane-dev-v4-worker": [
  {
   "paths": [
    {
     "age-nanoseconds": 138734627807,
     "family": {
      "afi": "ipv4",
      "safi": "unicast"
     },
     "nlri": {
      "base64": "GAoBAQ=="
     },
     "path-attributes": [
      {
       "base64": "QAEBAA=="
      },
      {
       "base64": "QAIGAgEAAP3q"
      },
      {
       "base64": "QAMECgACAg=="
      }
     ]
    }
   ],
   "prefix": "10.1.1.0/24",
   "router-asn": 65002
  }
 ]
}
```

### CODE REVIEW

File: bgp/routes.go [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+			if err != nil {
+				continue
+			}
+			for _, path := range r.Paths {
+				fmt.Fprintf(w, "%d\t", route.RouterAsn)
+				fmt.Fprintf(w, "%s\t", path.NLRI)
+				fmt.Fprintf(w, "%s\t", nextHopFromPathAttributes(path.PathAttributes))
+				fmt.Fprintf(w, "%s\t", time.Duration(path.AgeNanoseconds).Round(time.Second))
+				fmt.Fprintf(w, "%s\t", path.PathAttributes)
+				fmt.Fprintf(w, "\n")
```
Comment (harsimran-pabla) [inline bgp/routes.go:264]: Should new line be added if len(path) = 0 and also at line 256 ?

**Review by harsimran-pabla — APPROVED** [review]
Thank you, LGTM.

File: bgp/routes.go (hunk 1 above)
Comment (rastislavs) [reply bgp/routes.go:264]: that would be kind of an invalid route, but can do that just in case

File: bgp/routes.go [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+		} else {
+			allFetchedData[fetchedData.nodeName] = fetchedData.data
+		}
+	}
+
+	return allFetchedData, err
+}
+
+func (s *Status) fetchRoutesFromPod(ctx context.Context, args []string, pod *corev1.Pod) ([]*models.BgpRoute, error) {
+	cmd := []string{"cilium-dbg", "bgp", "routes"}
```
Comment (derailed) [inline bgp/routes.go:124]: Do we need to recompute the static cmd for every pods?

**Review by derailed — COMMENTED** [review]
@rastislavs Nice work!

File: bgp/routes.go (hunk 2 above)
Comment (rastislavs) [reply bgp/routes.go]: good point, fixed

**Review by derailed — APPROVED** [review]
@rastislavs Thank you for the updates!

**Comment by rastislavs** [comment]
@sayboras could you please take a look at this PR? It is now missing only `vendor` review.

**Review by sayboras — APPROVED** [review]
LGTM :heavy_check_mark:

**Pull request merged**